beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_NDT_ELLIPSOID_HPP
+#define BELUGA_ROS_NDT_ELLIPSOID_HPP
+
+#include <beluga_ros/messages.hpp>
+
+/**
+ * \file
+ * \brief Utilities for NDT I/O over ROS interfaces.
+ */
+
+namespace beluga_ros {
+
+template <typename RealScalar>
+inline bool isApprox(RealScalar x, RealScalar y, RealScalar prec = Eigen::NumTraits<RealScalar>::dummy_precision()){
+  return std::abs(x-y) < prec;
+}
+
+inline bool use_mean_covariance(auto& eigenSolver, const auto& cell, auto& marker){
+  eigenSolver.compute(cell.covariance);
+  if(eigenSolver.info() != Eigen::Success){
+    return false;
+  }
+  
+  // Compute the rotation based on the covariance matrix
+  const auto eigenvectors = eigenSolver.eigenvectors().real();
+  const auto eigenvalues = eigenSolver.eigenvalues();
+
+  Eigen::Matrix3d rotationMatrix;
+  Eigen::Vector3d scalevector;
+  rotationMatrix << eigenvectors.col(0), eigenvectors.col(1), eigenvectors.col(2);
+  scalevector = Eigen::Vector3d{eigenvalues.x(), eigenvalues.y(), eigenvalues.z()};
+
+  // Permutation
+  if(!isApprox(rotationMatrix.determinant(), 1.0)) {
+    rotationMatrix << eigenvectors.col(1), eigenvectors.col(0), eigenvectors.col(2);
+    scalevector = Eigen::Vector3d{eigenvalues.y(), eigenvalues.x(), eigenvalues.z()};
+  }
+
+  if(!isApprox(rotationMatrix.determinant(), 1.0)) {
+    return false;
+  }
+  const auto rotation = Eigen::Quaterniond{rotationMatrix};
+
+  // Fill in the message
+  marker.type = beluga_ros::msg::Marker::SPHERE;
+  marker.action = beluga_ros::msg::Marker::ADD;
+
+  marker.pose.position.x = cell.mean[0];
+  marker.pose.position.y = cell.mean[1];
+  marker.pose.position.z = cell.mean[2];
+
+  marker.pose.orientation.x = rotation.x();
+  marker.pose.orientation.y = rotation.y();
+  marker.pose.orientation.z = rotation.z();
+  marker.pose.orientation.w = rotation.w();
+
+  float scaleConstant = 2.0f;
+  marker.scale.x = std::sqrt(scalevector.x()) * scaleConstant;
+  marker.scale.y = std::sqrt(scalevector.y()) * scaleConstant;
+  marker.scale.z = std::sqrt(scalevector.z()) * scaleConstant;
+
+  return true;
+}
+
+inline void use_cell_size(const auto& position, const auto& size, auto& marker){
```

Comment:
@Fernando-Sanz I sense this needs some formatting, are we using [`pre-commit` hooks](https://github.com/Ekumen-OS/beluga/blob/main/CONTRIBUTING.md#how-do-i-submit-a-good-code-contribution)?
```suggestion
inline void use_cell_size(const auto& position, const auto& size, auto& marker) {
```

Reply:
Yes(?). Now I'm not sure, I installed it and it was working with previous commits. I'm checking it.

Reply:
Linter are happy now though https://github.com/Ekumen-OS/beluga/actions/runs/12319843458/job/34482145756?pr=453.

So I guess this isn't needed anymore